yt_dlp/extractor/bilibili.py

```diff
@@ -86,10 +88,30 @@ def extract_formats(self, play_info):
         } for video in traverse_obj(play_info, ('dash', 'video', ...)))
 
         missing_formats = format_names.keys() - set(traverse_obj(formats, (..., 'quality')))
-        if missing_formats:
+        if formats and missing_formats:
             self.to_screen(f'Format(s) {", ".join(format_names[i] for i in missing_formats)} are missing; '
                            f'you have to login or become premium member to download them. {self._login_hint()}')
 
+        fragments = traverse_obj(play_info, ('durl', lambda _, v: url_or_none(v['url']), {
+            'url': ('url', {url_or_none}),
+            'duration': ('length', {lambda x: float_or_none(x, scale=1000)}),
+            'filesize': ('size', {int_or_none}),
+        }))
+        if fragments:
+            formats.append({
+                'url': fragments[0]['url'],
+                'filesize': sum(traverse_obj(fragments, (..., 'filesize'))),
+                **({
+                    'fragments': fragments,
+                    'protocol': 'http_dash_segments'
```

Comment:
The patch by GD_Slime tried using `http_dash_segments` to handle segmented multi-video flv, which #6438 previously tried to address.
It appears that `http_dash_segments` downloader is not meant for this and will cause timestamp problems, and later converting this to a `multi_video` playlist (and discarding other formats) would be necessay, unless we have a downloader that downloads and merges segments just like `multi_video`.